mycore-base/src/main/java/org/mycore/access/MCRAccessManager.java

```diff
@@ -65,11 +65,20 @@
         MCRShutdownHandler.getInstance().addCloseable(EXECUTOR_SERVICE::shutdownNow);
     }
 
+    @SuppressWarnings("unchecked")
     public static <T extends MCRAccessInterface> T getAccessImpl() {
-        return (T) MCRConfiguration2.getSingleInstanceOf("MCR.Access.Class", MCRAccessBaseImpl.class).get();
+        return (T) MCRConfiguration2.<MCRAccessInterface>getInstanceOf("MCR.Access.Class")
+            .orElseGet(MCRAccessBaseImpl::new);
     }
 
     private static MCRAccessCheckStrategy getAccessStrategy() {
+        // if acccessStrategy equals accessImp we reuse the accessImpl, 
+        // to make sure, that only one singleton gets created
+        // (used to instantiate fact-based access system) 
+        if (MCRConfiguration2.getString("MCR.Access.Strategy.Class").orElse("UNDEFINED")
```

Comment:
This doesnt work if the Strategy is defined in MCR.Access.Strategy (which is the preferred way).

Reply:
Also "UNDEFINED" makes no sense since you can check if the optional is null.

Reply:
reimplemented property comparison